model/src/main/kotlin/io/spine/validation/required/RequiredPolicy.kt

```diff
@@ -53,28 +67,32 @@ internal class RequiredPolicy : ValidationPolicy<FieldExited>() {
             name = fieldName
             type = declaringType
         }
-        val field = select(RequiredField::class.java).findById(id)
+        val field = select<RequiredField>().findById(id)
         if (field != null && field.required) {
             val declaration = findField(fieldName, declaringType, event.file)
             val rule = requiredRule(declaration, field)
             return rule.asA()
         }
         return ignore()
     }
-}
 
-private fun requiredRule(declaration: Field, field: RequiredField): RuleAdded {
-    val rule = RequiredRule.forField(declaration, field.errorMessage)
-        ?: throwDoesNotSupportRequired(declaration)
-    return rule.toEvent(declaration.declaringType)
-}
+    private fun requiredRule(declaration: Field, field: RequiredField): RuleAdded {
+        val rule = RequiredRule.forField(declaration, field.errorMessage)
+            ?: throwDoesNotSupportRequired(declaration)
+        return rule.toEvent(declaration.declaringType)
+    }
 
-private fun throwDoesNotSupportRequired(field: Field): Nothing {
-    val fieldName = field.name.value
-    val typeUrl = field.declaringType.typeUrl
-    val type: PrimitiveType = field.type.primitive
-    error(
-        "The field `${typeUrl}.${fieldName}` of the type `${type}`" +
-                " does not support `($REQUIRED)` validation.",
-    )
+    private fun throwDoesNotSupportRequired(field: Field): Nothing {
+        val file = field.declaringFile(typeSystem!!)
```

Comment:
We moved functions back under the class because now we need the `typeSystem` property.